sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/ParentElement.java

```diff
@@ -42,4 +51,106 @@ default void updateParameters(Map<String, String> parameters) {
             child.updateParameters(parameters);
         }
     }
+
+    @Override
+    default boolean isEligible(Collection<String> dirtyAssets) {
+        return Element.super.isEligible(dirtyAssets) && !isInterrupted();
+    }
+
+    /** Updates children */
+    @Override
+    default void update(Protos.TaskStatus taskStatus) {
+        Collection<? extends Element> children = getChildren();
+        LOGGER.info("Updated {} with TaskStatus: {}", getName(), TextFormat.shortDebugString(taskStatus));
+        children.forEach(element -> element.update(taskStatus));
+    }
+
+    /** Restarts children. */
+    @Override
+    default void restart() {
+        Collection<? extends Element> children = getChildren();
+        LOGGER.info("Restarting elements within {}: {}", getName(), children);
+        children.forEach(element -> element.restart());
+    }
+
+    /** Force completes children. */
+    @Override
+    default void forceComplete() {
+        Collection<? extends Element> children = getChildren();
+        LOGGER.info("Forcing completion of elements within {}: {}", getName(), children);
+        children.forEach(element -> element.forceComplete());
+    }
+
+    /**
+     * Returns all errors from this {@Link Element} and all its children.
+     *
+     * @param parentErrors Errors from this {@Link Element} itself.
+     * @return a combined list of all errors from the parent and all its children.
+     */
+    default List<String> getErrors(List<String> parentErrors) {
+        // Note that this function MUST NOT call parent.getErrors() as that creates a circular call.
+        List<String> errors = new ArrayList<>(); // copy list to avoid modifying parentErrors in-place
+        errors.addAll(parentErrors);
+        Collection<? extends Element> children = getChildren();
+        children.forEach(element -> errors.addAll(element.getErrors()));
+        return errors;
+    }
+
+    @Override
+    default Status getStatus() {
+        // Ordering matters throughout this method.  Modify with care.
+        // Also note that this function MUST NOT call parent.getStatus() as that creates a circular call.
```

Comment:
This second note (on L102) can go away. The equivalent under the new system would be `getStatus()` calling `this.getStatus()` -- the result there shouldn't be surprising for anyone.